kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/Schemas.java

```diff
@@ -0,0 +1,214 @@
+  private static final String RESOURCE_URI_SCHEME = "resource";
+
+  public static Schema fromAvsc(InputStream in) throws IOException {
+    // the parser has state, so use a new one each time
+    return new Schema.Parser().parse(in);
+  }
+
+  public static Schema fromAvsc(File location) throws IOException {
+    return fromAvsc(
+        FileSystem.getLocal(DefaultConfiguration.get()),
+        new Path(location.getPath()));
+  }
+
+  public static Schema fromAvsc(FileSystem fs, Path path) throws IOException {
+    InputStream in = null;
+    boolean threw = true;
+
+    try {
+      in = fs.open(path);
+      Schema schema = new Schema.Parser().parse(in);
+      threw = false;
+      return schema;
+    } finally {
+      Closeables.close(in, threw);
+    }
+  }
+
+  public static Schema fromAvsc(Configuration conf, URI location)
+      throws IOException {
+    InputStream in = null;
+    boolean threw = true;
+
+    try {
+      in = open(conf, location);
+      Schema schema = fromAvsc(in);
+      threw = false;
+      return schema;
+    } finally {
+      Closeables.close(in, threw);
+    }
+  }
+
+  public static Schema fromAvro(InputStream in) throws IOException {
+    GenericDatumReader<GenericRecord> datumReader =
+        new GenericDatumReader<GenericRecord>();
+    DataFileStream<GenericRecord> stream = null;
+    boolean threw = true;
+
+    try {
+      stream = new DataFileStream<GenericRecord>(in, datumReader);
+      Schema schema = stream.getSchema();
+      threw = false;
+      return schema;
+    } finally {
+      Closeables.close(stream, threw);
+    }
+  }
+
+  public static Schema fromAvro(File location) throws IOException {
+    return fromAvro(
+        FileSystem.getLocal(DefaultConfiguration.get()),
+        new Path(location.getPath()));
+  }
+
+  public static Schema fromAvro(FileSystem fs, Path location)
+      throws IOException {
+    InputStream in = null;
+    boolean threw = true;
+
+    try {
+      in = fs.open(location);
+      Schema schema = fromAvro(in);
+      threw = false;
+      return schema;
+    } finally {
+      Closeables.close(in, threw);
+    }
+  }
+
+  public static Schema fromAvro(Configuration conf, URI location)
+      throws IOException {
+    InputStream in = null;
+    boolean threw = true;
+
+    try {
+      in = open(conf, location);
+      Schema schema = fromAvro(in);
+      threw = false;
+      return schema;
+    } finally {
+      Closeables.close(in, threw);
+    }
+  }
+
+  public static Schema fromParquet(File location) throws IOException {
+    return fromParquet(
+        FileSystem.getLocal(DefaultConfiguration.get()),
+        new Path(location.getPath()));
+  }
+
+  public static Schema fromParquet(FileSystem fs, Path location) throws IOException {
+    ParquetMetadata footer = ParquetFileReader.readFooter(fs.getConf(), location);
+
+    String schemaString = footer.getFileMetaData()
+        .getKeyValueMetaData().get("parquet.avro.schema");
```

Comment:
We will need to wait for 1.6 to reliably use Parquet's properties. Some versions used parquet.avro.schema and some used avro.schema. The current version, 1.5, only has a property for parquet.avro.schema because it doesn't have the compatibility for the older property. When the version with OLD_PARQUET_READ_SCHEMA is released we can update this.